### PR TITLE
add jenkisn api

### PR DESCRIPTION
**Pull request by soulseen**
@runzexia

### CODE REVIEW

File: pkg/apiserver/devops/devops.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package devops
+
+import (
+	"github.com/emicklei/go-restful"
+	"kubesphere.io/kubesphere/pkg/errors"
+	"kubesphere.io/kubesphere/pkg/models/devops"
+	"net/http"
+)
+
+const HOST = "http://ks-jenkins.kubesphere-devops-system.svc.cluster.local:80"
```
Comment (runzexia) [inline pkg/apiserver/devops/devops.go]: Use flag to pass in the address of jenkins, and it is convenient to debug locally.

**Comment by runzexia** [comment]
@zryfish @wansir

File: pkg/apiserver/devops/devops.go (hunk 1 above)
Comment (soulseen) [reply pkg/apiserver/devops/devops.go]: ok,already updated.

File: pkg/models/devops/devops.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	err = json.Unmarshal(resBody, &res)
+
+	return res, err
+}
+
+// create jenkins request
+func jenkinsClient(baseUrl string, req *http.Request) ([]byte, error) {
+	newReqUrl, err := url.Parse(baseUrl)
+	if err != nil {
+		return nil, err
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: log error

File: pkg/models/devops/devops.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	newRequest := &http.Request{
+		Method: req.Method,
+		URL:    newReqUrl,
+		Header: req.Header,
+		Body:   req.Body,
+	}
+
+	client := &http.Client{Timeout: 30 * time.Second}
+	resp, err := client.Do(newRequest)
+	if err != nil {
```
Comment (zryfish) [inline pkg/models/devops/devops.go]: same here

File: pkg/models/devops/urlconfig.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+ */
+package devops
+
+
+// Some apis for Jenkins.
+const (
```
Comment (zryfish) [inline pkg/models/devops/urlconfig.go]: format code before committing

**Pull request closed without merging**